docs/source/configuration/traffic-shaping.mdx

```diff
@@ -44,7 +33,135 @@ traffic_shaping:
+      capacity: 10
+      interval: 5s # Must not be greater than 18_446_744_073_709_551_615 milliseconds and not less than 0 milliseconds
+```
+
+### Experimental request retry
 
+On failure, subgraph requests can be retried automatically. This is deactivated by default for mutations.
+This uses [Finagle's *RetryBudget* algorithm](https://finagle.github.io/blog/2016/02/08/retry-budgets/),
+in which every successful request adds an expirable token to a bucket, and every retry consumes a number
+of those tokens. On top of that, a minimal number of retries per second is available, to test regularly
+when the retry budget was entirely consumed or on startup when very few requests have been sent.
+The tokens expire so the budget has a large number of available retries if a lot of recent requests were
+successful, but reduces quickly on frequent failures to avoid sending too much traffic to the subgraph.
+
+It is configurable as follows:
+
+```yaml title="router.yaml"
+traffic_shaping:
+  all:
+    experimental_retry:
+      min_per_sec: 10 # minimal number of retries per second (`min_per_sec`, default is 10 retries per second)
+      ttl: 10s # for each successful request, we register a token, that expires according to this option (default: 10s)
+      retry_percent: 0.2 # defines the proportion of available retries to the current number of tokens
+      retry_mutations: false # allows retries on mutations. This should only be enabled if mutations are idempotent
 ```
 
-Any configuration under the `subgraphs` key takes precedence over configuration under the `all` key. In the example above, query deduplication is enabled for all subgraphs _except_ the `products` subgraph.
+### Variable deduplication
+
+In federated queries, using the `_entities` operation, it is often the case that one entity is requested from
+multiple parts of a query. As an example, for each review of each product, geting the author's name might return
+the same name multiple times if an author wrote multiple reviews.
+
+To reduce the size of subgraph requests and the amount of work they might perform, the list of entities sent
+can be deduplicated. This is activated with the `deduplicate_variables` option:
+
+```yaml title="router.yaml"
+traffic_shaping:
+  deduplicate_variables: true # Enable the variable deduplication optimization.
+```
+
+### Query deduplication
+
+If similar queries are sent to a subgraph, the Apollo Router can avoid sending them multiple times, and instead
+copy the result of the first query when it comes back from the subgraph, thus reducing the traffic to the subgraph
+and the overall client request latency.
+
+The subgraph queries have to be exactly the same (path, headers and body) to be deduplicated.
+
+```yaml title="router.yaml"
+traffic_shaping:
+  all:
+    deduplicate_query: true # Enable query deduplication for all subgraphs.
+```
+
+### Ordering
```

Comment:
If this means "ordering in the configuration", then this is really important and should be right at the top of the page with a little more text to explain why.

If it doesn't mean that, then what's the purpose of this section?

Reply:
This is to explain in which order the different steps are applied, so we can understand how they interact. The configuration order will have no impact on it.

Reply:
Ok. In that case, perhaps re-word that sentence to be more clear about why we are telling people this. Something like:

```
Traffic shaping will alway be executed in this order within the router in order to ensure that interactions between different steps are correct. Declaration order in configuration does not affect this ordering.
```

Reply:
fixed in 4e3f567b